src/inmanta/module.py

```diff
@@ -2342,3 +2572,92 @@ def get_all_python_requirements_as_list(self) -> List[str]:
 
     def get_module_requirements(self) -> List[str]:
         return [str(req) for req in self.get_module_v2_requirements()]
+
+    def add_module_requirement(self, requirement: InmantaModuleRequirement, add_as_v1_module: bool) -> None:
+        if add_as_v1_module:
+            raise Exception("Cannot add V1 requirement to a V2 module")
+        # Parse config file
+        config_parser = ConfigParser()
+        config_parser.read(self.get_metadata_file_path())
+        python_pkg_requirement: Requirement = requirement.to_python_package_requirement()
+        if config_parser.has_option("options", "install_requires"):
+            new_install_requires = [
+                r
+                for r in config_parser.get("options", "install_requires").split("\n")
+                if r and Requirement.parse(r).key != python_pkg_requirement.key
+            ]
+            new_install_requires.append(str(python_pkg_requirement))
+        else:
+            new_install_requires = [str(python_pkg_requirement)]
+        config_parser.set("options", "install_requires", "\n".join(new_install_requires))
+        # Write config back to disk
+        with open(self.get_metadata_file_path(), "w", encoding="utf-8") as fd:
+            config_parser.write(fd)
+        # Reload in-memory state
+        with open(self.get_metadata_file_path(), "r", encoding="utf-8") as fd:
+            self._metadata = ModuleV2Metadata.parse(fd)
+
+    def has_module_requirement(self, module_name: str) -> bool:
+        config_parser = ConfigParser()
+        config_parser.read(self.get_metadata_file_path())
+        if not config_parser.has_option("options", "install_requires"):
+            return False
+        pkg_name: str = ModuleV2.get_package_name_for(module_name)
+        return any(
+            r
+            for r in config_parser.get("options", "install_requires").split("\n")
+            if r and Requirement.parse(r).key == pkg_name
+        )
+
+
+class YamlMetadataFileWithRequiresField:
+    @classmethod
+    def add_module_requirement_and_write(
```

Comment:
The fact that all methods on this class are class methods with as first actual argument a union of classes, makes me itch to add this functionality to a shared superclass (or mixin class) of Project and ModuleV1.

But I'm not certain if that would be better. 

I could e.g. be a class like `ModuleLikeWithRequirementTxt`?

Reply:
(by adding a supercall, the add_module_requirement_and_write could even stack nicely with the one for setup.cfg and modules.yaml, but that may cause other problems)